### PR TITLE
Init containers in prow job pods: handle running but not ready case

### PR DESCRIPTION
**Pull request by chaodaiG**
Fixes https://github.com/kubernetes/test-infra/issues/21985

### CODE REVIEW

File: prow/spyglass/lenses/metadata/lens.go [hunk 1]
```diff
@@ -249,8 +249,9 @@ func hintFromPodInfo(buf []byte) string {
 			msg = fmt.Sprintf("state: terminated, reason: %q, message: %q", state.Reason, state.Message)
 		} else if state := ic.State.Waiting; state != nil {
 			msg = fmt.Sprintf("state: waiting, reason: %q, message: %q", state.Reason, state.Message)
-		} else if state := ic.State.Running; state != nil { // This shouldn't happen at all, just in case.
-			logrus.WithField("pod", report.Pod.Name).WithField("container", ic.Name).Warning("Init container is running but not ready")
+		} else if state := ic.State.Running; state != nil {
+			// Yes this is weird, but it did happened https://github.com/kubernetes/test-infra/issues/21985
+			msg = "state: running"
```
Comment (alvaroaleman) [inline prow/spyglass/lenses/metadata/lens.go:254]: Isn't this essentially a bug in kube?

Comment (chaodaiG) [reply prow/spyglass/lenses/metadata/lens.go:254]: feels like so, created https://github.com/kubernetes/kubernetes/issues/101585

**Review by cjwagner — APPROVED** [review]

**Pull request merged**